ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,152 @@
+
+  // for smart contract interactions, methodData can be used to derive the name of the action being taken
+  const methodData = useSelector((state) => getKnownMethodData(state, initialTransaction?.txParams?.data)) || {}
+
+  const actionKey = getTransactionActionKey(initialTransaction)
+  const statusKey = getStatusKey(primaryTransaction)
+  const status = statusKey in pendingStatusMap ? PENDING_STATUS : statusKey
+
+  const primaryValue = primaryTransaction.txParams?.value
+  let prefix = '-'
+  const date = formatDateWithYearContext(initialTransaction.time || 0)
+  let subtitle
+  let recipientAddress = to
+
+  const token = knownTokens.find((token) => token.address === recipientAddress)
+  const tokenData = useTokenData(initialTransaction?.txParams?.data)
+  const tokenDisplayValue = useTokenDisplayValue(initialTransaction?.txParams?.data, token)
+
+  let category
+  let title
+  // There are four types of transaction entries that are currently differentiated in the design
+  // 1. (PENDING DESIGN) signature request
+  // 2. Send (sendEth sendTokens)
+  // 3. Deposit
+  // 4. Site interaction
+  // 5. Approval
+  if (transactionCategory == null) {
+    const origin = initialTransaction.msgParams?.origin || initialTransaction.origin
+    category = TRANSACTION_CATEGORY_SIGNATURE_REQUEST
+    title = t('signatureRequest')
+    subtitle = origin || ''
+  } else if (transactionCategory === APPROVED_STATUS) {
+    category = TRANSACTION_CATEGORY_APPROVAL
+    title = t('approveSpendLimit')
+    subtitle = initialTransaction.origin
+  } else if (transactionCategory === DEPLOY_CONTRACT_ACTION_KEY || transactionCategory === CONTRACT_INTERACTION_KEY) {
+    category = TRANSACTION_CATEGORY_INTERACTION
+    title = (methodData?.name && camelCaseToCapitalize(methodData.name)) || (actionKey && t(actionKey)) || ''
+    subtitle = initialTransaction.origin
+  } else if (transactionCategory === INCOMING_TRANSACTION_KEY) {
+    category = TRANSACTION_CATEGORY_RECEIVE
+    title = t('receive')
+    prefix = ''
+    subtitle = `${t('from')}: ${shortenAddress(senderAddress)}`
+  } else if (transactionCategory === TOKEN_METHOD_TRANSFER || transactionCategory === TOKEN_METHOD_TRANSFER_FROM) {
+    category = TRANSACTION_CATEGORY_SEND
+    title = t('sendSpecifiedTokens', [token?.symbol])
+    recipientAddress = getTokenToAddress(tokenData.params)
+    subtitle = `${t('to')}: ${shortenAddress(recipientAddress)}`
+  } else if (transactionCategory === SEND_ETHER_ACTION_KEY) {
+    category = TRANSACTION_CATEGORY_SEND
+    title = t('sendETH')
+    subtitle = `${t('to')}: ${shortenAddress(recipientAddress)}`
+  }
+
+  const primaryCurrencyPreferences = useUserPreferencedCurrency(PRIMARY)
+  const secondaryCurrencyPreferences = useUserPreferencedCurrency(SECONDARY)
+
+  const [primaryCurrency] = useCurrencyDisplay(primaryValue, {
+    prefix,
+    displayValue: tokenDisplayValue,
+    suffix: token?.symbol,
```

Comment:
Similar to the above comment; we should make sure not to include this for non-transfer transactions

Reply:
Though the `secondaryCurrency` might be the one requiring a suffix depending on the user's preference, so the `suffix` might be required on the `secondaryCurrency` instead of the `primaryCurrency` 🤔

Reply:
The secondaryCurrency will never display a token value unless I'm mistaken...

Reply:
Right, but that's the problem. Or at least it seems like a problem - maybe it's intentional 🤷‍♂️.

`useUserPreferencedCurrency` determines whether fiat is "primary" or "secondary", and from there the `displayValue` overrides that to show the token value. So something is wonky here; right now the primary will always show the token, and the secondary will show either fiat or eth depending on user's preferred currency. But it'll show whichever one _isn't_ their preferred currency. So a user that prefers ETH will be shown fiat as secondary, and vice versa.

I'm not sure what we want to do here 🤔 

Either:

1) Show the token value in whichever place ETH would normally be shown
2) Show the token value first, with the user's preferred currency second
3) Show the token value first, with the user's non-preferred currency second
4) Show the token value alone.

Reply:
I may be confused -- is this something that differs from the current develop branch or are we discussing an improvement that shows relative eth value or fiat value for the token?

Reply:
The current develop branch doesn't show the token value at all here. I thought you had implemented #3 at the time I wrote that comment, which seems kinda wrong. But since your latest update, it's #4, which seems just fine for the time being :+1:

Reply:
Er... I guess the original comment here is still not resolved, though the tangential topic I brought up re: the secondary currency is.

I think this should still depend upon `isTokenCategory` being true:

```suggestion
    suffix: isTokenCategory && token?.symbol,
```

Reply:
Could we jump on a call today at some point to walk through this? I’m still not following what is unintended and what the issue is with this logic

Reply:
Sure! Though, in a screenshot:

![dia-with-fiat](https://user-images.githubusercontent.com/2459287/82904675-2ae7ad80-9f39-11ea-8c4e-2a97586b9708.PNG)

The current designs show a fiat secondary balance. It's unclear to me whether this was supposed to _always_ show fiat, but it's certainly different from this PR which omits the secondary balance altogether.

Reply:
Right now I'm going to leave it as always and only showing the token value. I'll follow up with @rachelcope about whether it was intended to show fiat currency for token transactions, but if we do that it can't use our existing logic and would be a hefty lift because we don't have (to my knowledge) exchange rates for tokens yet? If we showed Fiat here would it be the current going rate for those tokens or would it be the rate at the time of purchase?

Reply:
We do have exchange rates for tokens in the send flow, and probably other places as well. It was never included in the transaction list though.

I agree that omitting this for now makes sense :+1: